src/MyHealth/symptoms.ts

```diff
@@ -22,3 +45,81 @@ export const determineHealthAssessment = (
   }
   return HealthAssessment.NotAtRisk
 }
+
+type WithADate = { date: Posix }
+const compareDatesAscending = (
+  { date: dateLeft }: WithADate,
+  { date: dateRight }: WithADate,
+) => {
+  return dateLeft < dateRight ? -1 : 1
+}
+const compareDatesDescending = (
+  { date: dateLeft }: WithADate,
+  { date: dateRight }: WithADate,
+) => {
+  return dateLeft < dateRight ? 1 : -1
+}
+
+const groupLogEntriesDaily = (allEntries: SymptomLogEntry[]): LogDataByDay => {
+  const groupedEntries: LogDataByDay = {}
+
+  allEntries.forEach((entry) => {
+    const { date } = entry
+    const entryDateBeginningOfDay = beginningOfDay(date)
+
+    if (groupedEntries[entryDateBeginningOfDay]) {
+      const newLogEntries = groupedEntries[entryDateBeginningOfDay].logEntries
+      newLogEntries.push(entry)
+      newLogEntries.sort(compareDatesAscending)
+      groupedEntries[entryDateBeginningOfDay].logEntries = newLogEntries
+    } else {
+      groupedEntries[entryDateBeginningOfDay] = {
+        logEntries: [entry],
+        checkIn: null,
+      }
+    }
+  })
+
+  return groupedEntries
+}
+
+const joinLogDataAndDailyCheckIns = (
+  logData: LogDataByDay,
```

Comment:
Thoughts on calling this `logDataByDay`? Maybe it's just me but I'm getting a bit turned around with the types.

```suggestion
  logDataByDay: LogDataByDay,
```